src/Peachpie.Library.Network/CURLConstants.cs

```diff
@@ -1067,11 +1086,71 @@ public override void Apply(Context ctx, HttpWebRequest request)
 
         private static void LoadCookieFile(HttpWebRequest request, PhpStream stream)
         {
-            // TODO: load netscape-like or header-style cookies from stream
+            // load netscape-like or header-style cookies from stream
 
-            // request.CookieContainer.Add() // add parsed cookie
+            // TODO: add support for header-style cookies (`Set-Cookie: ...`).
+            //       currently, only curl's netscape-like cookies are supported.
 
             // request.CookieContainer.SetCookies( ... ) // set header style cookies
+
+            string line = String.Empty;
+
+            // iterate over the cookie jar lines
+            while (!stream.Eof)
+            {
+
+                // read a line, and strip off the possible end-of-line characters
+                line = stream.ReadLine(-1, null).Trim();
+
+                // ignore blank lines
+                if (String.IsNullOrWhiteSpace(line))
+                    continue;
+
+                bool httpOnly;
+
+                // check for HttpOnly cookies and remove the #HttpOnly prefix
+                if (line.Length < 10 || line.Substring(0, 10) != "#HttpOnly_")
+                    httpOnly = false;
+
+                else
+                {
+                    line = line.Substring(10);
+                    httpOnly = true;
+                }
+
+                // we only look for non-comment, non-blank, valid cookies per lines
+                if (!line.StartsWith("#") && line.Count(c => c == '\t') == 6)
+                {
+                    // get tokens in an array and trim them
+                    string[] tokens = line.Split('\t').Select(token => token.Trim()).ToArray();
+
+                    // gather current cookie information in a new cookie object
+                    var cookie = new Cookie();
+
+                    cookie.Domain = tokens[0].TrimStart('.'); // The domain that created AND can read the variable.
+                    // var subdomainAccess = bool.Parse(tokens[1]); // boolean value indicating if all machines within a given domain can access the variable.
+                    cookie.Path = tokens[2] ?? "/"; // The path within the domain that the variable is valid for.
+                    cookie.Secure = bool.Parse(tokens[3]); // boolean value indicating if a secure connection with the domain is needed to access the variable.
+
+                    // normalize cookie path according to curl behavior
+                    cookie.Path = cookie.Path.TrimEnd('/') + '/';
```

Comment:
- should be `"/"` (string), `/` causes unnecessary allocation and conversion to string
- recommending to use `PathUtils.TrimEndSeparator(cookie.Path)`, faster without additional allocations

Reply:
@jakubmisek I believe `PathUtils.TrimEndSeparator()` implementation will only remove the last present `/` character, whereas `TrimEnd()` will remove as many as `/` characters present:
https://github.com/peachpiecompiler/peachpie/blob/6e8d6649a3110c6f6b640ad862515e699d5f214b/src/Peachpie.Runtime/Utilities/PathUtils.cs#L28-L33
Should this behavior be changed? Consider the following string:

```cs
string somePath = "/some/path///"
```

Reply:
you are right, so please keep `TrimEnd` (if it's what curl does)

Reply:
@jakubmisek I'll stick to `TrimEndSeparator()` since cookie behaves [the same](https://github.com/curl/curl/blob/9c1806ae4684ec5ef1aeb39bb9f15cece1c27256/lib/cookie.c#L326-L329) way.